4.0 rc1: Not enough warnings on i386 in test_data_out_of_range

When running on i386, I get the following failures due to missing warnings:
```
_______________ test_data_out_of_range[False-fast_reader2-True] ________________

parallel = False, fast_reader = {'parallel': False, 'use_fast_converter': True}
guess = True

    @pytest.mark.parametrize("guess", [True, False])
    # fast_reader configurations: False| 'use_fast_converter'=False|True
    @pytest.mark.parametrize('fast_reader', [False, dict(use_fast_converter=False),
                                             dict(use_fast_converter=True)])
    @pytest.mark.parametrize("parallel", [False, True])
    def test_data_out_of_range(parallel, fast_reader, guess):
        """
        Numbers with exponents beyond float64 range (|~4.94e-324 to 1.7977e+308|)
        shall be returned as 0 and +-inf respectively by the C parser, just like
        the Python parser.
        Test fast converter only to nominal accuracy.
        """
        # Python reader and strtod() are expected to return precise results
        rtol = 1.e-30
    
        # Update fast_reader dict; adapt relative precision for fast_converter
        if fast_reader:
            fast_reader['parallel'] = parallel
            if fast_reader.get('use_fast_converter'):
                rtol = 1.e-15
            elif np.iinfo(np.int_).dtype == np.dtype(np.int32):
                # On 32bit the standard C parser (strtod) returns strings for these
                pytest.xfail("C parser cannot handle float64 on 32bit systems")
    
        if parallel:
            if not fast_reader:
                pytest.skip("Multiprocessing only available in fast reader")
            elif TRAVIS:
                pytest.xfail("Multiprocessing can sometimes fail on Travis CI")
    
        fields = ['10.1E+199', '3.14e+313', '2048e+306', '0.6E-325', '-2.e345']
        values = np.array([1.01e200, np.inf, np.inf, 0.0, -np.inf])
        with catch_warnings(AstropyWarning) as w:
            t = ascii.read(StringIO(' '.join(fields)), format='no_header',
                           guess=guess, fast_reader=fast_reader)
        if fast_reader and not parallel:  # Assert precision warnings for cols 2-5
>           assert len(w) == 4
E           assert 3 == 4
E            +  where 3 = len([<warnings.WarningMessage object at 0xe8d488b0>, <warnings.WarningMessage object at 0xe8d48538>, <warnings.WarningMessage object at 0xe8d48c10>])

astropy/io/ascii/tests/test_c_reader.py:1067: AssertionError
```
The same happens for `False-fast_reader2-False`.

--- a/astropy/io/ascii/src/tokenizer.c
+++ b/astropy/io/ascii/src/tokenizer.c
@@ -1003,7 +1003,7 @@ double xstrtod(const char *str, char **endptr, char decimal,
         else if (exponent < 0)
             number /= e[-exponent];
 
-        if (number == HUGE_VAL || number == -HUGE_VAL)
+        if (number >= HUGE_VAL || number <= -HUGE_VAL)
             errno = ERANGE;
     }
 

